src/promise-btn.directive.ts

```diff
@@ -33,12 +33,21 @@ export class PromiseBtnDirective implements OnDestroy, AfterContentInit {
   }
 
   @Input()
-  set promiseBtn(promise: any) {
-    if (promise instanceof Observable) {
-      this.promise = promise.toPromise();
-    } else {
-      this.promise = promise;
+  set promiseBtn(request: any) {
+    const isObservable: boolean = request instanceof Observable;
+    const isPromise: boolean = request instanceof Promise || (
+      request !== null &&
+      typeof request === 'object' &&
+      typeof request.then === 'function' &&
+      typeof request.catch === 'function'
```

Comment:
this check makes sense as in initPromiseHandler we call both then and catch functions
```javascript
      promise
        .then(resolveLoadingState)
        .catch(resolveLoadingState);
```